sound/soc/sof/sof-acpi-dev.c

```diff
@@ -214,12 +240,13 @@ static int sof_acpi_probe(struct platform_device *pdev)
 	/* set callback to enable runtime_pm */
 	sof_pdata->sof_probe_complete = sof_acpi_probe_complete;
 #endif
-	/* call sof helper for DSP hardware probe */
-	ret = snd_sof_device_probe(dev, sof_pdata);
-	if (ret) {
-		dev_err(dev, "error: failed to probe DSP hardware!\n");
+	dev_set_drvdata(dev, sof_pdata);
+
+	/* add MFD devices */
+	ret = devm_mfd_add_devices(dev, -1, &sof_dsp_device, 1, NULL, 0,
+				   NULL);
```

Comment:
can you pass pdata to the MFD devices instead of using an indirection to call the parents?

Reply:
it's actually what you do for PCI but not here?